### PR TITLE
enh(perf): set LDAP contactgroup synchronization every hour

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

The LDAP contactgroup synchronization was made every minute.
Replace this value to 60minutes instead

**Fixes** # (none)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [x] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: cron/centAcl.php [hunk 1]
```diff
@@ -110,9 +111,29 @@
...
+        switch ($row['key']) {
+            case 'ldap_auth_enable':
+                $ldap_enable = $row['value'];
+                break;
+            case 'ldap_last_acl_update':
+                $ldap_last_update = $row['value'];
+                break;
+        }
+    }
+    if ($ldap_enable == 1 && $ldap_last_update < (time() - LDAP_UPDATE_PERIOD)) {
```
Comment (kduret) [inline cron/centAcl.php]: $ldap_enable and $ldap_last_update are not initalized if no one ldap is configured
Furthermore, you should use camelcase for variable name

File: cron/centAcl.php [hunk 2]
```diff
@@ -110,9 +111,29 @@
         programExit($errorMessage);
     }
 
-    $resourceCache = array();
+    /**
+     * Sync ACL with LDAP's contactgroup
+     * If the LDAP is enable and the last check is greater than the update period
```
Comment (adr-mo) [inline cron/centAcl.php]: Just to be niggling but there is small English error on the comment. "enable" => "enabled"

File: cron/centAcl.php (hunk 1 above)
Comment (adr-mo) [reply cron/centAcl.php]: Same comments as @kduret

File: cron/centAcl.php [hunk 3]
```diff
@@ -110,9 +111,29 @@
...
+     *
+     * @TODO : Synchronize LDAP with contacts data in background to avoid it at login
+     */
+    $queryOptions = "SELECT `key`, `value` FROM `options` WHERE `key` IN ('ldap_auth_enable', 'ldap_last_acl_update')";
```
Comment (sc979) [inline cron/centAcl.php]:
```suggestion
    $ldap_enable = $ldap_last_update = 0;
    $queryOptions = "SELECT `key`, `value` FROM `options` WHERE `key` IN ('ldap_auth_enable', 'ldap_last_acl_update')";
```

File: cron/centAcl.php [hunk 4]
```diff
@@ -110,9 +111,30 @@
         programExit($errorMessage);
     }
 
-    $resourceCache = array();
+    /**
+     * Sync ACL with LDAP's contactgroup
+     * If the LDAP is enable and the last check is greater than the update period
```
Comment (sc979) [inline cron/centAcl.php]:
```suggestion
     * If the LDAP is enabled and the last check is greater than the update period
```

**Pull request merged**